ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1194,6 +1194,34 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
             break;
 #endif
 
+        case MAV_CMD_DO_WINCH:
+            // param1 : winch number (ignored)
+            // param2 : action (0=release, 1=length control, 2=rate control). See WINCH_ACTIONS enum.
+            if(!copter.g2.winch.enabled()) {
+                result = MAV_RESULT_FAILED;
+            } else {
+                result = MAV_RESULT_ACCEPTED;
+                switch ((uint8_t)packet.param2) {
+                    case WINCH_DISABLE:
+                        copter.g2.winch.relax();
+                        break;
+                    case WINCH_LENGTH_CONTROL: {
+                        float desired_length = copter.g2.winch.get_line_length() + packet.param3;
+                        copter.g2.winch.release_length(desired_length);
```

Comment:
How come there are not any Log_Write_Event's here for length/rate on the winch, but their are in the mission? (This is where the later out of scope comment would be nice)

Reply:
Txs, added log-write-event messages when winch is operated via MAVlink.